src/core/modules/SolarSystem.cpp

```diff
@@ -537,72 +537,43 @@ void SolarSystem::updateSkyCulture(const StelSkyCulture& skyCulture)
 	planetNativeNamesMap.clear();
 	planetNativeNamesMeaningMap.clear();
 
-	QString namesFile = skyCulture.path + "/planet_names.fab";
+	if (!skyCulture.names.isEmpty())
+		loadCultureSpecificNames(skyCulture.names);
 
-	if (!QFileInfo(namesFile).exists())
-	{
-		for (const auto& p : std::as_const(systemPlanets))
-		{
-			if (p->getPlanetType()==Planet::isPlanet || p->getPlanetType()==Planet::isMoon || p->getPlanetType()==Planet::isStar)
-			{
-				p->setNativeName("");
-				p->setNativeNameMeaning("");
-			}
-		}
-		updateI18n();
-		return;
-	}
-
-	// Open file
-	QFile planetNamesFile(namesFile);
-	if (!planetNamesFile.open(QIODevice::ReadOnly | QIODevice::Text))
-	{
-		qDebug() << " Cannot open file" << QDir::toNativeSeparators(namesFile);
-		return;
-	}
-
-	// Now parse the file
-	// lines to ignore which start with a # or are empty
-	static const QRegularExpression commentRx("^(\\s*#.*|\\s*)$");
-
-	// lines which look like records - we use the RE to extract the fields
-	// which will be available in recRx.capturedTexts()
-	static const QRegularExpression recRx("^\\s*(\\w+)\\s+\"(.+)\"\\s+_[(]\"(.+)\"[)]\\n");
-
-	QString record, planetId, nativeName, nativeNameMeaning;
+	updateI18n();
+}
 
-	// keep track of how many records we processed.
-	int totalRecords=0;
-	int readOk=0;
-	int lineNumber=0;
-	while (!planetNamesFile.atEnd())
+void SolarSystem::loadCultureSpecificNames(const QJsonObject& data)
+{
+	for (auto it = data.begin(); it != data.end(); ++it)
 	{
-		record = QString::fromUtf8(planetNamesFile.readLine());
-		lineNumber++;
+		const auto key = it.key();
+		const auto specificNames = it->toArray();
 
-		// Skip comments
-		if (commentRx.match(record).hasMatch())
-			continue;
-
-		totalRecords++;
-
-		QRegularExpressionMatch match=recRx.match(record);
-		if (!match.hasMatch())
+		if (key.startsWith("NAME "))
 		{
-			qWarning() << "ERROR - cannot parse record at line" << lineNumber << "in planet names file" << QDir::toNativeSeparators(namesFile);
-		}
-		else
-		{
-			planetId          = match.captured(1).trimmed();
-			nativeName        = match.captured(2).trimmed();
-			nativeNameMeaning = match.captured(3).trimmed();
+			const auto planetId = key.mid(5);
+			const auto names = it.value().toArray();
+
+			// We store only one name per planet, so just take the first valid one
+			// FIXME: maybe we should somehow store all
```

Comment:
I am afraid yes. Think of Venus as Evening star and Morning star, these names could be even completely different in some SCs. (We could actually define rules per elongation, later, to select the appropriate name.) I think I have seen more than one name somewhere (Polynesian area?)

Reply:
I guess this should be left for a subsequent PR, since current `master` doesn't have any support for this either.

Reply:
> I guess this should be left for a subsequent PR, since current `master` doesn't have any support for this either.

I agree for a subsequent PR. After introducing this feature I've tried to add the support of multiple names for inner planets as function of longitude, but solution was complicated and a bit "hackish" - I just postponed the task in later time and forget about it.